thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -1297,10 +1309,13 @@ namespace diskann {
         if (bitset_view.empty() || !bitset_view.test(frontier_nhood.first)) {
           float cur_expanded_dist;
           if (!use_disk_index_pq) {
+            if (metric == Metric::COSINE) {
+              normalize(node_fp_coords_copy, aligned_dim);
+            }
```

Comment:
Looks like our logic is doing normalization for the base vector for every single search. Will this be time consuming? As discussed, we planned to store the norm of all base vectors during the building time to skip this part right?

Reply:
By benchmarking this will not cause performance problem